File: frontend/src/api/ChatAPI.ts
// Chat API wrapper for Physical AI Chatbot
interface QueryRequest {
  question: string;
  book_id: string;
  selected_text?: string | null;
}

interface QueryResponse {
  answer: string;
  sources: string[];
  mode_used: string;
}

// Legacy response format for compatibility
interface LegacyQueryResponse {
  response: string;
  sources: string[];
  sessionId: string;
  timestamp: string;
}

class ChatAPI {
  private baseUrl: string;

  constructor(baseUrl: string) {
    this.baseUrl = baseUrl;
  }

  async sendMessage(
    question: string,
    selectedText?: string
  ): Promise<LegacyQueryResponse> {
    const payload: QueryRequest = {
      question: question,
      book_id: "physical-ai-book",
      selected_text: selectedText || null,
    };

    const response = await fetch(`${this.baseUrl}/query`, {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify(payload),
    });

    if (!response.ok) {
      const error = await response.json().catch(() => ({}));
      throw new Error(error.detail || `HTTP ${response.status}`);
    }

    const data: QueryResponse = await response.json();

    // Convert to legacy format for compatibility with existing frontend
    return {
      response: data.answer,
      sources: data.sources,
      sessionId: `session-${Date.now()}`,
      timestamp: new Date().toISOString(),
    };
  }
}

export default ChatAPI;
export type { QueryRequest, QueryResponse };

File: backend/src/api/chat_api.py
"""
Simple chat API endpoints
"""
from fastapi import APIRouter, Request
from pydantic import BaseModel
from typing import List, Optional

router = APIRouter()

class ChatRequest(BaseModel):
    question: str
    book_id: str
    selected_text: Optional[str] = None

class ChatResponse(BaseModel):
    answer: str
    sources: List[str]
    mode_used: str

@router.post("/chat", response_model=ChatResponse)
async def chat_endpoint(request: ChatRequest, req: Request):
    """Simple chat endpoint"""
    answer = f"You asked: '{request.question}'. This is a response from the Physical AI chatbot system."
    return ChatResponse(
        answer=answer,
        sources=["book-content"],
        mode_used="full_book"
    )
